benchmark/single-source/StringBreadcrumbs.swift

```diff
@@ -0,0 +1,494 @@
+  }
+
+  override func tearDown() {
+    super.tearDown()
+    inputIndices = []
+  }
+
+  @inline(never)
+  override func run() {
+    outputOffsets.removeAll(keepingCapacity: true)
+    for index in inputIndices {
+      outputOffsets.append(inputString._toUTF16Offset(index))
+    }
+  }
+}
+
+/// Split a string into `count` random slices and convert their UTF-16 offsets
+/// into String index ranges.
+class UTF16ToIndexRange: BenchmarkBase {
+  let count: Int
+  var inputOffsets: [Range<Int>] = []
+  var outputIndices: [Range<String.Index>] = []
+
+  init(workload: Workload, count: Int) {
+    self.count = count
+    super.init(name: "StringBreadcrumbs.UTF16ToIndexRange", workload: workload)
+  }
+
+  override func setUp() {
+    super.setUp()
+    var rng = LCRNG(seed: seed)
+    inputOffsets = (
+      0 ..< inputString.utf16.count
+    ).randomIndexRanges(count: count, using: &rng)
+    outputIndices = []
+    outputIndices.reserveCapacity(inputOffsets.count)
+  }
+
+  override func tearDown() {
+    super.tearDown()
+    inputOffsets = []
+  }
+
+  @inline(never)
+  override func run() {
+    outputIndices.removeAll(keepingCapacity: true)
+    for range in inputOffsets {
+      outputIndices.append(inputString._toUTF16Indices(range))
+    }
+  }
+}
+
+/// Split a string into `count` random slices and convert their index ranges
+/// into into UTF-16 offset pairs.
+class IndexToUTF16Range: BenchmarkBase {
+  let count: Int
+  var inputIndices: [Range<String.Index>] = []
+  var outputOffsets: [Range<Int>] = []
+
+  init(workload: Workload, count: Int) {
+    self.count = count
+    super.init(name: "StringBreadcrumbs.IndexToUTF16Range", workload: workload)
+  }
+
+  override func setUp() {
+    super.setUp()
+    var rng = LCRNG(seed: seed)
+    inputIndices = self.inputString.randomIndexRanges(count: count, using: &rng)
+    outputOffsets = []
+    outputOffsets.reserveCapacity(inputIndices.count)
+  }
+
+  override func tearDown() {
+    super.tearDown()
+    inputIndices = []
+  }
+
+  @inline(never)
+  override func run() {
+    outputOffsets.removeAll(keepingCapacity: true)
+    for range in inputIndices {
+      outputOffsets.append(inputString._toUTF16Offsets(range))
+    }
+  }
+}
+
+
+class CopyUTF16CodeUnits: BenchmarkBase {
+  let count: Int
+  var inputIndices: [Range<Int>] = []
+  var outputBuffer: [UInt16] = []
+
+  init(workload: Workload, count: Int) {
+    self.count = count
+    super.init(name: "StringBreadcrumbs.CopyUTF16CodeUnits", workload: workload)
+  }
+
+  override func setUp() {
+    super.setUp()
+    var rng = LCRNG(seed: seed)
+    inputIndices = (
+      0 ..< inputString.utf16.count
+    ).randomIndexRanges(count: count, using: &rng)
+    outputBuffer = Array(repeating: 0, count: inputString.utf16.count)
+  }
+
+  override func tearDown() {
+    super.tearDown()
+    inputIndices = []
+  }
+
+  @inline(never)
+  override func run() {
+    for range in inputIndices {
+      print(range)
```

Comment:
This line is breaking the benchmark run. From https://ci.swift.org/job/swift-PR-osx-perf/607/console:
````
16:41:04 583,StrToInt,3,1615,1668,1633,30,1616
16:41:04 584,StringAdder,3,461,461,461,0,461
16:41:04 2303..<2325
16:41:04 3435..<3454
16:41:04 4487..<4501
16:41:04 1556..<1580
16:41:04 1124..<1141
````

Reply:
My bad. The crash was due to PR 20768 not being merged at the time it was run.
But still… that debug print needs to go! 😝

Reply:
Oops, d'oh...